pyccel/parser/semantic.py

```diff
@@ -1381,14 +1382,18 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, **settings):
 
                 # Not yet supported for arrays: x=y+z, x=b[:]
                 # Because we cannot infer shape of right-hand side yet
-                know_lhs_shape = lhs.shape or (lhs.rank == 0) \
+                know_lhs_shape = all(sh is not None for sh in lhs.alloc_shape) \
+                        or (lhs.rank == 0) \
                         or isinstance(rhs, (Variable, EmptyLike, DottedVariable))
                 if not know_lhs_shape:
                     msg = "Cannot infer shape of right-hand side for expression {} = {}".format(lhs, rhs)
                     errors.report(PYCCEL_RESTRICTION_TODO+'\n'+msg,
                         bounding_box=(self._current_fst_node.lineno, self._current_fst_node.col_offset),
                         severity='fatal', blocker=self.blocking)
-
+                elif lhs.rank>0 and isinstance(rhs, PyccelOperator):
+                    #TODO: Provide order once issue #335 is fixed
+                    #stmts.append(Assign(lhs, Empty(lhs.alloc_shape, dtype, lhs.order)))
+                    new_expressions.append(Assign(lhs, Empty(lhs.alloc_shape, dtype, 'C')))
```

Comment:
Should we fix issue #335 before merging this PR?

Reply:
I don't know how long it would take to fix that issue. This PR looks good as it stands

Reply:
I would rather leave it for another PR. I was just worried about potentially introducing a bug here